master/static/migrations/20240409101349_add-arbitrary-metadata-table.tx.up.sql

```diff
@@ -0,0 +1,16 @@
+-- ALTER RUNS TABLE
+ALTER TABLE runs ADD COLUMN metadata JSONB;
```

Comment:
This is a product question.

I'm worried the feature is really complicated and going to be hard to understand for users. Can metadata be a ```map[string]string``` or even just a ```[]string```? I'm curious the use case behind needing to be able to specify nested fields.

Reply:
From Piyush:
> We have feedback from customers, AMLEs, and compete signals that this is what customers generally expect from a metadata logging capability.
> As for the nested fields, they provide users with the ability to organize the metadata, params, configs etc. in the way that they want and provide the right flexibility in being able to log more complex metadata structures.

I could add better documentation around expectation for the use of this feature. My understanding is that it's to capture information about a specific trial/run that can only be known at run time.